qutebrowser/html/tabs.html

```diff
@@ -0,0 +1,56 @@
+{% extends "styled.html" %}
+
+{% block style %}
+{{super()}}
+h1 {
+    margin-bottom: 10px;
+}
+
+.url a {
+    color: #444;
+}
+
+th {
+    text-align: left;
+}
+
+.qmarks .name {
+    padding-left: 5px;
+}
+
+.empty-msg {
+    background-color: #f8f8f8;
+    color: #444;
+    display: inline-block;
+    text-align: center;
+    width: 100%;
+}
+{% endblock %}
+
+{% block content %}
+
+<h1>Tab list</h1>
+{% for win_id, tabs in tab_list_by_window.items() %}
+<h2>Window {{ win_id }}</h2>
+<table class="tabs_win{{win_id}}">
+    <tbody>
+    {% for name, url in tabs %}
+    <tr>
+        <td class="name"><a href="{{url}}">{{name}}</a></td>
+        <td class="url"><a href="{{url}}">{{url}}</a></td>
+    </tr>
+    {% endfor %}
+    </tbody>
+</table>
+{% endfor %}
+
+<h2>Raw list</h2>
```

Comment:
Hmm, you mentioned you needed this for a userscript - I wonder, does it make sense to display them there? As the user doesn't care about the raw list, maybe hide it?

Reply:
It is not for a userscript (in the sense of Greasemonkey), but a completely separate script which takes a newline-separated list of URLs as input. In order to run this from a userscript, the javascript code should be able to spawn a process separate process on my computer (unless I am mistaken, this is not possible).

Would it be OK to put it in a `<details>` tag ?

Reply:
I meant an [userscript](https://github.com/qutebrowser/qutebrowser/blob/master/doc/userscripts.asciidoc) as in a qutebrowser-userscript (which inherited the concept and name from [dwb](https://bitbucket.org/portix/dwb)).

With that, you could read `$QUTE_HTML` and get the data from there, even if it's not visible in the rendered HTML. And you could easily trigger that with a keybinding.

Reply:
@dopsi Any update on this?

Reply:
@The-Compiler I'm currently having exams, but I think I'll have time over the weekend to look at it.

Reply:
I know that feeling - good luck with your exams! If you don't get to it on the weekend, don't worry.